### PR TITLE
BigQueryDynamicTableFactory unit test case added

### PR DESCRIPTION
**Pull request by Sridhar-Durgam2022**

### CODE REVIEW

File: src/test/java/com/google/cloud/flink/bigquery/DefaultDynamicTableContext.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ */
+package com.google.cloud.flink.bigquery;
+
+import java.util.Map;
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.table.catalog.CatalogTableImpl;
+import org.apache.flink.table.catalog.ObjectIdentifier;
+import org.apache.flink.table.factories.DynamicTableFactory;
+
+public class DefaultDynamicTableContext implements DynamicTableFactory.Context {
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/DefaultDynamicTableContext.java:24]: Should be `MockDynamicTableContext` as it is not the default implementation

File: src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+public class BigQueryDynamicTableFactoryTest {
+  @Test
+  public void factoryIdentifierTestSuccess() {
+    BigQueryDynamicTableFactory bigQueryDynamicTableFactory = new BigQueryDynamicTableFactory();
+    String factoryIdentfier = bigQueryDynamicTableFactory.factoryIdentifier();
+    String expectedFactoryIdentifier = "bigquery";
+    assertThat(factoryIdentfier).isNotNull();
+    assertThat(factoryIdentfier).isEqualTo(expectedFactoryIdentifier);
+    assertTrue(factoryIdentfier instanceof String);
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java:53]: redundant as it is checked by the compiler

File: src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    assertTrue(bigQueryDynamicTableFactory instanceof DynamicTableSourceFactory);
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java:54]: redundant as it is checked by the compiler

File: src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+  @Test
+  public void requiredOptionsTestSuccess() {
+
+    BigQueryDynamicTableFactory bigQueryDynamicTableFactory = new BigQueryDynamicTableFactory();
+    List<String> expectedOptions =
+        Arrays.asList(
+            "materializationDataset", "materializationProject", "query", "selectedFields", "table");
+    Set<ConfigOption<?>> configOptionSet = bigQueryDynamicTableFactory.requiredOptions();
+    assertThat(configOptionSet).isNotNull();
+    assertTrue(configOptionSet instanceof Set);
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java:66]: redundant as it is checked by the compiler

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java [hunk 5]
```diff
@@ -163,20 +156,14 @@ public DynamicTableSource createDynamicTableSource(Context context) {
...
-  @Override
-  public DynamicTableSink createDynamicTableSink(Context context) {
-    return null;
-  }
-
   private LinkedList<String> getReadStreamNames(ReadableConfig options) {
     bigQueryReadClientFactory = null;
     UserAgentHeaderProvider userAgentHeaderProvider;
     BigQueryCredentialsSupplier bigQueryCredentialsSupplier;
     LinkedList<String> readStreamNames = new LinkedList<String>();
```
[the remark this answers is not in the document]
Comment (davidrabinowitz) [reply src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java:163]: @Sridhar-Durgam2022 any reason to use LinkedList?

File: src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java [hunk 6]
```diff
@@ -136,16 +128,17 @@ public Set<ConfigOption<?>> optionalOptions() {
     options.add(BQ_BACKGROUND_THREADS_PER_STREAM);
     options.add(PARALLELISM);
     options.add(MAX_PARALLELISM);
+    options.add(ARROW_COMPRESSION_CODEC);
     return options;
   }
 
   DecodingFormat<DeserializationSchema<RowData>> decodingFormat;
+  private String arrowReadSessionSchema;
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java]: Addressed

File: src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java]: Addressed

[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java]: Addressed, Changed to ArrayList

File: src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java]: Addressed

File: src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java (hunk 2 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java]: Removed

File: src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java (hunk 4 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java:65]: Removed

File: src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java]: Addressed

File: src/main/java/com/google/cloud/flink/bigquery/FlinkBigQueryConfig.java [hunk 7]
```diff
@@ -776,85 +783,19 @@ public int expirationTimeInMinutes() {
...
-      return formatOptions;
-    }
-
-    public String getFileSuffix() {
-      return getFormatOptions().getType().toLowerCase();
-    }
-  }
-
-  private static class ParseSqlString {
+  class SqlParser {
```
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/FlinkBigQueryConfig.java:786]: Addressed

File: src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java]: Addressed

File: src/main/java/com/google/cloud/flink/bigquery/FlinkBigQueryConfig.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (Sridhar-Durgam2022) [reply src/main/java/com/google/cloud/flink/bigquery/FlinkBigQueryConfig.java:786]: Addressed

File: src/test/java/com/google/cloud/flink/bigquery/DefaultDynamicTableContext.java (hunk 1 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/DefaultDynamicTableContext.java]: Addressed

File: src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+
+    BigQueryDynamicTableFactory bigQueryDynamicTableFactory = new BigQueryDynamicTableFactory();
+    List<String> expectedOptions =
+        Arrays.asList(
+            "materializationDataset", "materializationProject", "query", "selectedFields", "table");
+    Set<ConfigOption<?>> configOptionSet = bigQueryDynamicTableFactory.requiredOptions();
+    assertThat(configOptionSet).isNotNull();
+    assertThat(configOptionSet.size()).isEqualTo(5);
+    List<String> options = new ArrayList<String>();
+    Iterator<ConfigOption<?>> configOptionIterator = configOptionSet.iterator();
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java:65]: Replace lines 64-70 with `List<String> options = configOptionSet.stream().map(configOption -> configOption.key().toString()).collect(Collectors.toString());`

File: src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+    while (configOptionIterator.hasNext()) {
+      ConfigOption configOption = configOptionIterator.next();
+      options.add(configOption.key().toString());
+    }
+    Collections.sort(options);
+    assertTrue(expectedOptions.equals(options));
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java:71]: replace with `assertThat(expectedOptions).containsAll(options);`

File: src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+            "partitionType",
+            "proxyPassword",
+            "proxyUri",
+            "proxyUsername");
+    List<String> options = new ArrayList<String>();
+    BigQueryDynamicTableFactory bigQueryDynamicTableFactory = new BigQueryDynamicTableFactory();
+    Set<ConfigOption<?>> optionalOptions = bigQueryDynamicTableFactory.optionalOptions();
+    assertThat(optionalOptions).isNotNull();
+    assertThat(optionalOptions.size()).isEqualTo(18);
+    optionalOptions.forEach(
```
Comment (davidrabinowitz) [inline src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java:101]: update the method like `requiredOptionsTestSuccess()`

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java (hunk 10 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java:98]: renamed

File: src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java (hunk 9 above)
Comment (Sridhar-Durgam2022) [reply src/test/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactoryTest.java]: Replaced with assertThat(expectedOptions).containsExactlyElementsIn(options);

**Pull request closed without merging**